native/src/binaryRead.cpp

```diff
@@ -2174,13 +2174,13 @@ void searchMapData(CodedInputStream* input, MapRoot* root, MapIndex* ind, Search
 	}
 }
 
-void convertRouteDataObjecToMapObjects(SearchQuery* q, std::vector<RouteDataObject*>& list,
+void convertRouteDataObjecToMapObjects(SearchQuery* q, std::vector<SHARED_PTR<RouteDataObject>>& list,
 									   std::vector<FoundMapDataObject>& tempResult, int& renderedState) {
-	std::vector<RouteDataObject*>::iterator rIterator = list.begin();
```

Comment:
raw pointers usually better for performance, so this requires deep investigation only if it's necessary - this code is used in a critical area